sklearn/manifold/t_sne.py

```diff
@@ -747,7 +753,8 @@ def _fit(self, X, skip_num_points=0):
 
             # Find the nearest neighbors for every point
             knn = NearestNeighbors(algorithm='auto', n_neighbors=k,
-                                   metric=self.metric)
+                                   metric=self.metric,
+                                   metric_params=self.metric_params)
```

Comment:
I think `NearestNeighbors` expects `p` not to be in `metric_params`. Around [here](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/neighbors/base.py#L112)